app/uk/gov/hmrc/taxenrolmentassignmentfrontend/models/UsersGroupResponse.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.taxenrolmentassignmentfrontend.models
+
+import play.api.libs.json.{Format, Json}
+
+case class AdditonalFactors(factorType: String,
+                            phoneNumber: Option[String] = None,
+                            name: Option[String] = None)
+
+case class UsersGroupResponse(obfuscatedUserId: String,
```

Comment:
What is the different between UserGroupResponse and AccountDetails, do we need both?

Reply:
The way the UsersGroupResponse is formatted isn't suitable for our needs (date/times different, additionalFactors). There was one of two approaches that could be taken.
1. Write a complicated json reads which would need to validate we have received the correct json then format it into a model that suits us (I did start with this approach but it was getting really complicated and difficult to follow as every field needs to be extracted from the JsObject, cast to a type and then formatted to so it works with our views).
2. Do the validation of what we have received from users-group-search using default jsonReads, then cast it to a new model which has it in the correct format for our pages.

Both approaches work but with option 1 it was more complex and visually harder to follow.

Reply:
I think the extra case class is better. It can then also have things like enrolments added to it to avoid multiple calls to the cache to get the details around an account.